shipper.go

```diff
@@ -14,11 +14,220 @@ import (
+
+		if !isBlockDir(bd) {
+			continue
+		}
+
+		byt, err := ioutil.ReadFile(filepath.Join(dirname, bd.Name(), "meta.json"))
+		if err != nil {
+			level.Error(logger).Log("msg", "reload tempdir", "error", err.Error())
+			os.Exit(1)
+		}
+
+		var bm tsdb.BlockMeta
+		if err := json.Unmarshal(byt, &bm); err != nil {
+			level.Error(logger).Log("msg", "reload tempdir", "error", err.Error())
+			os.Exit(1)
+		}
+
+		if bm.Compaction.Level == 1 {
+			blocks = append(blocks, filepath.Join(dirname, bd.Name()))
+		}
+	}
+	return blocks
+}
+
+func lazyShip(dataDir string, lazyDataDir string, logger log.Logger, done <-chan bool) {
+
+	watcher, err := fsnotify.NewWatcher()
+	defer watcher.Close()
+	if err != nil {
+		level.Error(logger).Log("msg", "lazy Ship", "error", err.Error())
+	}
+	watcher.Add(dataDir)
+
+	ticker := time.NewTicker(5 * time.Second)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case event := <-watcher.Events:
+
+			if event.Op&fsnotify.Create == fsnotify.Create {
+
+				bd, err := os.Stat(event.Name)
+				if os.IsNotExist(err) {
+					break
+				} else if err != nil {
+					level.Error(logger).Log("msg", "lazy Ship", "error", err.Error())
+				}
+
+				if !isBlockDir(bd) {
+					break
+				}
+
+				byt, err := ioutil.ReadFile(filepath.Join(event.Name, "meta.json"))
+				if err != nil {
+					level.Error(logger).Log("msg", "lazy Ship", "error", err.Error())
+					break
+				}
+
+				var bm tsdb.BlockMeta
+				if err := json.Unmarshal(byt, &bm); err != nil {
+					level.Error(logger).Log("msg", "lazy Ship", "error", err.Error())
+				}
+
+				if bm.Compaction.Level != 1 {
+					break
+				}
+
+				lazyBlockDir := filepath.Join(lazyDataDir, bd.Name())
+				if err := os.MkdirAll(lazyBlockDir, 0777); err != nil {
+					level.Error(logger).Log("msg", "lazy Ship", "error", err.Error())
+				}
+
+				lazyChunksDir := filepath.Join(lazyBlockDir, "chunks")
+				if err := os.MkdirAll(lazyChunksDir, 0777); err != nil {
+					level.Error(logger).Log("msg", "lazy Ship", "error", err.Error())
+				}
+
+				// Hardlink meta, index and tombstones
+				for _, fname := range []string{
+					metaSuffix,
+					indexSuffix,
+					//tombstoneFilename,
+				} {
+					if err := os.Link(filepath.Join(event.Name, fname), filepath.Join(lazyBlockDir, fname)); err != nil {
+						level.Error(logger).Log("msg", "lazy Ship", "error", err.Error())
+					}
+				}
+
+				// Hardlink chunks
+				ChunkDir := filepath.Join(event.Name, "chunks")
+				files, err := ioutil.ReadDir(ChunkDir)
+				if err != nil {
+					level.Error(logger).Log("msg", "lazy Ship", "error", err.Error())
+				}
+
+				for _, f := range files {
+					err := os.Link(filepath.Join(ChunkDir, f.Name()), filepath.Join(lazyChunksDir, f.Name()))
+					if err != nil {
+						level.Error(logger).Log("msg", "lazy Ship", "error", err.Error())
+					}
+				}
+
+				/*ticker.Stop()
+				ticker = time.NewTicker(5 * time.Second)*/
+
+			}
+		case <-ticker.C:
+			blockDirs, err := ioutil.ReadDir(dataDir)
```

Comment:
Isn't this a repitition of `refresh()` defined above?

Reply:
refresh() returns all the valid block paths. Checking if the block path is valid or not is common but here we hardlink them to a temporary directory instead of returning them. May be a function to check the block is valid or not will reduce the repetition?